components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/recovery/v1/impl/core/PasswordRecoveryService.java

```diff
@@ -0,0 +1,480 @@
+import org.wso2.carbon.identity.rest.api.user.recovery.v1.model.RecoveryChannelInformation;
+import org.wso2.carbon.identity.rest.api.user.recovery.v1.model.RecoveryRequest;
+import org.wso2.carbon.identity.rest.api.user.recovery.v1.model.ResendConfirmationCodeExternalResponse;
+import org.wso2.carbon.identity.rest.api.user.recovery.v1.model.ResendConfirmationCodeInternalResponse;
+import org.wso2.carbon.identity.rest.api.user.recovery.v1.model.ResendConfirmationRequest;
+import org.wso2.carbon.identity.rest.api.user.recovery.v1.model.ResetCodeResponse;
+import org.wso2.carbon.identity.rest.api.user.recovery.v1.model.ResetRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.Response;
+
+/**
+ * Call internal OSGI services to perform password recovery.
+ */
+public class PasswordRecoveryService {
+
+    private static final Log log = LogFactory.getLog(PasswordRecoveryService.class.getName());
+
+    /**
+     * Initiate Password Recovery from POST.
+     *
+     * @param initRequest {@link InitRequest} object which holds the information in the account recovery
+     *                    request
+     * @return Response
+     */
+    public Response initiatePasswordRecovery(InitRequest initRequest) {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        Map<String, String> userClaims = RecoveryUtil.buildUserClaimsMap(initRequest.getClaims());
+        try {
+            // Get password recovery notification information.
+            RecoveryInformationDTO recoveryInformationDTO =
+                    UserAccountRecoveryServiceDataHolder.getPasswordRecoveryManager().initiate(userClaims, tenantDomain,
+                            RecoveryUtil.buildPropertiesMap(initRequest.getProperties()));
+            // If RecoveryChannelInfoDTO is null throw not found error.
+            if (recoveryInformationDTO == null) {
+                if (log.isDebugEnabled()) {
+                    String message = "No recovery information for password recovery request";
+                    log.debug(message);
+                }
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+            }
+            return Response.ok().entity(buildPasswordRecoveryInitResponse(tenantDomain, recoveryInformationDTO))
+                    .build();
+        } catch (IdentityRecoveryClientException e) {
+            throw RecoveryUtil.handleClientExceptions(PasswordRecoveryService.class.getName(), tenantDomain,
+                    IdentityRecoveryConstants.PASSWORD_RECOVERY_SCENARIO, Util.getCorrelation(), e);
+        } catch (IdentityRecoveryException e) {
+            throw RecoveryUtil.buildInternalServerErrorResponse(PasswordRecoveryService.class.getName(),
+                    Constants.SERVER_ERROR, e.getErrorCode(), Util.getCorrelation(), e);
+        } catch (Throwable throwable) {
```

Comment:
Let's catch Exception only. No need to catch Throwable

Reply:
Fixed: 588ec0bc091b7cb914b65b21bf870bbace8b930c